tests/integration/test_materialize_mysql_database/materialize_with_ddl.py

```diff
@@ -460,6 +460,7 @@ def query_event_with_empty_transaction(clickhouse_node, mysql_node, service_name
     mysql_node.query("INSERT INTO test_database.t1(a) VALUES(2)")
     mysql_node.query("/* start */ commit /* end */")
 
+    check_query(clickhouse_node, "SHOW TABLES FROM test_database FORMAT TSV", "t1\n")
     check_query(clickhouse_node, "SELECT * FROM test_database.t1 ORDER BY a FORMAT TSV", "1\tBEGIN\n2\tBEGIN\n")
```

Comment:
Ok, but how does it fix the failure? `check_query` retries not only in case of wrong query result, but also in case of exception. So, if `SELECT * FROM test_database.t1 ORDER BY a FORMAT TSV` fails with `Table test_database.t1 doesn't exist`, then `check_query` retires the query. 

Also there are error messages with other table names (`test_table_1`, `test_table_2`) in the [report](https://clickhouse-test-reports.s3.yandex.net/18684/ef39e7c069a6b655bc59fe6cec5466867426fa63/integration_tests_(asan)/integration_run_test_materialize_mysql_database_test_py_2.txt.out.log), it's not only about `t1`.

Reply:
fixed.
@tavplubix you are right, the test failed because `enable_global_with_statement = true` https://github.com/ClickHouse/ClickHouse/blob/master/src/Interpreters/executeQuery.cpp#L458